### PR TITLE
[Naitei][Feature][Whenever] Monthly Incomes Report

### PR DESCRIPTION
**Pull request by TranMinhHuy2003**
## Related Tickets
- [ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/78443)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2024-08-19 14-52-15
Screenshot from 2024-08-19 14-51-46

### CODE REVIEW

**Comment by TranMinhHuy2003** [comment]
ready

File: config/schedule.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# end
+#
+# every 4.days do
+#   runner "AnotherModel.prune_old_records"
+# end
+
+# Learn more: http://github.com/javan/whenever
+set :output, "log/cron_log.log"
+every "0 0 1 * *" do
+  runner "AdminMailer.send_monthly_incomes_report.deliver_now"
```
Comment (ghost) [inline config/schedule.rb]: Em dùng rails task nhé

**Review by ghost — COMMENTED** [review]
cmt

**Comment by TranMinhHuy2003** [comment]
ready

**Review by ghost — COMMENTED** [review]
conflict rùi

**Comment by TranMinhHuy2003** [comment]
ready

File: app/models/bill.rb [hunk 2]
```diff
@@ -85,13 +85,19 @@ def calculate_subtotal
     bill_details.sum{|detail| detail.product.price * detail.quantity}
   end
 
+  def self.monthly_incomes_report
+    where(status: :completed,
+          created_at: 1.month.ago.beginning_of_month..1.month.ago.end_of_month)
```
Comment (ghost) [inline app/models/bill.rb]: viết scope cho where đi em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by TranMinhHuy2003** [comment]
ready

**Pull request merged**